### DISCUSSION TITLE
Geodataset with Planet imagery?

### DISCUSSION THREAD

**Discussion by sofstef** [torchgeo/torchgeo#1100] [Q&A]
Hi, I have some Planet imagery from Planetscope (3m near daily product) and Skysat (roughly 50cm from a tasked flyover) and hand labelled coordinates of objects I'm interested in in geojson files–I saw a few examples in the repo (NASA marine debris and Spacenet) where Planet imagery and corresponding masks are loaded in a Nongeodataset and I've followed those to create a combined dataset. There's a bunch of reasons why a Geodataset would be more useful thouhg–can anybody give some pointers for what it would take to treat this as a geodataset or a quick summary as to why Planet imagery isn't suitable to be loaded in this way? 

When I tried creating a separate raster dataset for the imagery and vector dataset for the masks I got a bunch of errors (happy to describe those if useful) so I reverted back to the nongeodataset but curious to know if it's worth trying to make things work with a geodataset!

**Comment by adamjstewart** [comment]
We would absolutely love to have a GeoDataset for Planet imagery!

> can anybody give some pointers for what it would take to treat this as a geodataset

We have a whole tutorial for this: https://torchgeo.readthedocs.io/en/stable/tutorials/custom_raster_dataset.html

> a quick summary as to why Planet imagery isn't suitable to be loaded in this way?

There is no reason why we can't make a GeoDataset for Planet, the only reason we don't yet have one is that no one has gotten around to making one.

> When I tried creating a separate raster dataset for the imagery and vector dataset for the masks I got a bunch of errors (happy to describe those if useful)

Yes, please! In fact, it may be easiest to create a draft PR with what you have so far. That way I can download an example image and debug it myself.

**Reply by sofstef** [reply]
Thanks for the quick reply! 

EDIT: I just opened the draft pull request.
